packages/react-native/UPGRADING.md

```diff
@@ -0,0 +1,279 @@
++ [Bugsnag start];
+```
+
+Or if you need to customize your configuration:
+
+```diff
+- BugsnagConfiguration *config = [BugsnagConfiguration new];
+- config.releaseStage = @"beta";
+- [BugsnagReactNative startWithConfiguration:config];
++ BugsnagConfiguration *config = [BugsnagConfiguration loadConfig];
++ config.releaseStage = @"beta";
++ [Bugsnag startWithConfiguration:config];
+```
+
+See the new [Bugsnag Cocoa configuration guide](https://docs.bugsnag.com/platforms/ios/configuration-options/) for all available configuration options.
+
+If you have set your API key in `Info.plist`, the format has changed from a top level string to a dictionary.
+
+```
+- <key>BugsnagAPIKey</key>
+- <string>YOUR_API_KEY</string>
++ <key>bugsnag</key>
++ <dict>
++     <key>apiKey</key>
++     <string>YOUR_API_KEY</string>
++ </dict>
+```
+
+##### Android
+
+Replace the following import which will likely be in your `MainApplication.java` file:
+
+```diff
+- import com.bugsnag.android.BugsnagReactNative;
++ import com.bugsnag.android.Bugsnag;
+```
+
+Inside your application's `onCreate` method, replace any Bugsnag initialization code with the updated version:
+
+```diff
+- Configuration config = new Configuration();
+- config.setReleaseStage("beta");
+- BugsnagReactNative.startWithConfiguration(this /* app context */, config);
++ Configuration config = Configuration.load(this);
++ config.setReleaseStage("beta");
+Bugsnag.start(this /* app context */, config);
+```
+
+###### JS
+
+The JS interface has been updated to match our new notifier spec standard – that means it looks the same as our JS notifier, and uses the same names and concepts as our other platforms.
+
+When the JS layer loads, __the native layers and notifiers must already have been configured__. It you attempt to initialize the JS layer without having first configured the native layer, it __will not work__. This is new behaviour – previously we allowed configuration of the native layer to be done lazily and automatically, but this meant native errors on startup could be missed.
+
+Since the JS layer hooks in to an already running native layer, there are fewer configuration options available. Most configuration must be done in the native layer, with only a small subset of JS-specific options allowed in the JS layer.
+
+```diff
+- import { Client } from 'bugsnag-react-native';
+- const client = new Client();
++ import Bugsnag from '@bugsnag/react-native'
++ Bugsnag.start()
+```
+
+JS configuration options are now supplied directly to the `start` method:
+
+```diff
+- import { Configuration } from 'bugsnag-react-native'
+- const config = new Configuration()
+- config.codeBundleId = '1.2.3-r091283'
+- const client = new Client(config)
++ Bugsnag.start({
++   codeBundleId: '1.2.3-r091283'
++ })
+```
+
+See the [React Native configuration options page](https//docs.bugsnag.com/platforms/react-native/react-native/configuration-options) for reference.
+
+#### General usage
+
+##### iOS
```

Comment:
The subheadings are getting quite nested here, to the extent that the bold body text stands out more than the heading.

Could we have an H5 referring to native usage followed by H5's for each of the JS sections?

Reply:
I think I have addressed this.